src/plonk/keygen.rs

```diff
@@ -168,29 +170,25 @@ where
 {
     let (domain, cs, config) = create_domain::<C, ConcreteCircuit>(params);
 
+    if (params.n as usize) < cs.minimum_rows() {
```

Comment:
There seem to be a lot of `.n as usize` (at least 32 instances). Should it instead be declared as `usize`? Not for this PR.

Reply:
Yeah, it's a general issue. I want the code to work on both 32-bit and 64-bit platforms, so I chose to use `u32` for some of the lengths to essentially force me to address the fact that some larger lengths are impossible by adding errors and explicit checks. Once the checks were written, though, it started to just get annoying to cast as `usize` all over the place. We can fix this later.